sound/soc/sof/intel/cnl.c

```diff
@@ -147,10 +147,12 @@ static int cnl_ipc_cmd_done(struct snd_sof_dev *sdev, int dir)
 
 static int cnl_ipc_is_ready(struct snd_sof_dev *sdev)
 {
-	u64 val;
+	u64 busy, done;
 
-	val = snd_sof_dsp_read(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDR);
-	if (val & CNL_DSP_REG_HIPCIDR_BUSY)
+	busy = snd_sof_dsp_read(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDR);
+	done = snd_sof_dsp_read(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDA);
+	if ((busy & CNL_DSP_REG_HIPCIDR_BUSY) ||
+	    (done & CNL_DSP_REG_HIPCIDA_DONE))
```

Comment:
(done & CNL_DSP_REG_HIPCIDA_DONE) means msg done, it is actually ready, you are using it opposite?

Reply:
no, it is right.  After DONE bit is set, the reture value of read would be zero.
if it is not set, the return value is 1. So it is different with we think.